kratos/processes/apply_multi_point_constraints_process.h

```
@@ -0,0 +1,333 @@
+    {
+        Parameters default_parameters(R"(
+            {
+                "constraint_set_name":"default",
+                "master_sub_model_part_name":"default_master",
+                "slave_sub_model_part_name":"default_slave",                
+                "variable_names":[],
+                "reform_every_step":false   
+            }  )");
+        mParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+        // IMPORTANT : This constructor is not to be used when using this process in the normal KRATOS process_list of python script
+        ProcessInfoPointerType info = mrModelPart.pGetProcessInfo();
+        if (!info->Has(CONSTRAINTS_CONTAINER))
+             info->SetValue(CONSTRAINTS_CONTAINER, Kratos::make_shared<std::vector<ConstraintPointerType>>());
+
+        pMpc = Kratos::make_shared<MultipointConstraint<SparseSpaceType,LocalSpaceType>>();
+        pMpc->SetName(rName);
+        pMpc->SetActive(true);
+
+        ConstraintSharedPointerVectorType mpc_data_vector = info->GetValue(CONSTRAINTS_CONTAINER);
+        (*mpc_data_vector).push_back(pMpc);
+    }
+
+    /**
+		Applies the MPC condition using two model parts, one as master and other as slave.
+        Here a nearest element interpolation is used by default to get the relation between master and slave
+		*/
+    void AddMasterSlaveRelation()
+    {
+        ModelPart &master_model_part = mrModelPart.GetSubModelPart(mParameters["master_sub_model_part_name"].GetString());
+        ModelPart &slave_model_part = mrModelPart.GetSubModelPart(mParameters["slave_sub_model_part_name"].GetString());
+
+        ProcessInfoPointerType info = mrModelPart.pGetProcessInfo();
+        int &dim = info->GetValue(DOMAIN_SIZE);
+
+        //std::string interpolationType = mParameters["interpolation_type"].GetString();
+        Parameters mapper_parameters = mParameters["interpolation_settings"];
+
+        if (dim == 2)
```

maybe throw an error in the else case?